csharp/test/Microsoft.ML.OnnxRuntime.Tests.NetCoreApp/InferenceTest.netcore.cs

```diff
@@ -686,10 +686,13 @@ private void TestRegisterCustomOpLibrary()
                 string libFullPath = Path.Combine(Directory.GetCurrentDirectory(), libName);
                 Assert.True(File.Exists(libFullPath), $"Expected lib {libFullPath} does not exist.");
 
-                var ortEnvInstance = OrtEnv.Instance();
-                string[] providers = ortEnvInstance.GetAvailableProviders();
-                if (Array.Exists(providers, provider => provider == "CUDAExecutionProvider")) {
-                    option.AppendExecutionProvider_CUDA(0);
+                using (var ortEnvInstance = OrtEnv.Instance())
```

Comment:
> using (var ortEnvInstance = OrtEnv.Instance()) [](http://example.com/codeflow?start=15&length=47)

It is a singleton, we never release it.

Reply:
I added the scoping of the `OrtEnv` in the tests to be able to create new instances using `ThreadingOptions`. Normally the class would be used as a singleton and throwing exceptions when attempting to create a new instance using `ThreadingOptions`.

I added a unit test to highlight this feature:

https://github.com/thoron/onnxruntime/blob/e30d55f9b51e44557b8c535f8a006f639d585037/csharp/test/Microsoft.ML.OnnxRuntime.Tests.Common/InferenceTest.cs#L273